process/block/shardblock.go

```diff
@@ -36,6 +37,9 @@ type shardProcessor struct {
 	currHighestMetaHdrNonce    uint64
 	allNeededMetaHdrsFound     bool
 
+	processedMiniBlocks    map[string][][]byte
```

Comment:
you could do map[string]map[string]bool - where you put in the second map the hashes, it is faster for constant searching

Reply:
and you have to clean it as well. as it would consume a lot of memory. use lru

Reply:
Refactored and added remove action